### PR TITLE
Fix: don't reference removed repeat instance as context for relevance computations

### LINKED ISSUE #908: FomLogicError thrown on relevance on field without a control element in a repeat
A form with cascade repeats throws the following error:

`FormLogicError: Could not evaluate: not(selected( ../uccmake ,'UccMakeOther')), message: Failed to execute 'evaluate' on 'XPathEvaluator': parameter 2 is not of type 'Node'.`

The issue is caused by a mismatch in the index of a cloned repeat node. When a repeat node is added or removed the `repeatIndex` doesn't seem to sync correctly with the `repeat-number` DOM element.

### PR DESCRIPTION
**Pull request by eyelidlessness**
Closes #908

### CODE REVIEW

**Comment by eyelidlessness** [comment]
Very probably addresses #908.

File: src/js/nodeset.js [hunk 1]
```diff
@@ -246,6 +246,7 @@ Nodeset.prototype.remove = function () {
                 nodes: null,
                 repeatPath,
                 repeatIndex,
+                removed: true,
```
Comment (lognaturel) [inline src/js/nodeset.js:249]: Hopefully this doesn't live that long. The event is marked as for internal use so it's not a big deal to add and remove from.

In the case of a visible form control, you can look at the parent and see if it exists. In the case of a node without a visible form control, without `removed`, you may know you're in some repeat instance when trying to compute cascade stuff but you don't know whether it's been deleted.

File: src/js/relevant.js [hunk 2]
```diff
@@ -203,8 +203,16 @@ export default {
...
             let context = p.path;
             if (
-                getChild(node, `.or-repeat-info[data-name="${p.path}"]`) &&
-                !getChild(node, `.or-repeat[name="${p.path}"]`)
+                (getChild(node, `.or-repeat-info[data-name="${p.path}"]`) &&
+                    !getChild(node, `.or-repeat[name="${p.path}"]`)) ||
+                // Repeat instance removed for model node with no visible form control
+                (insideRepeat && repeatParent == null && options.removed) ||
+                // All repeat instances removed for model node with no visible form control, i.e. during load with `jr:count="0"`
+                (repeatPath != null &&
```
Comment (lognaturel) [inline src/js/relevant.js:211]: Is `repeatPath != null` the same as `insideRepeat`?

File: src/js/relevant.js [hunk 3]
```diff
@@ -203,8 +203,16 @@ export default {
...
+                    repeatParent == null &&
```
Comment (lognaturel) [inline src/js/relevant.js:212]: `repeatParent == null` is only true if the node had no visible form control

File: src/js/relevant.js (hunk 2 above)
Comment (lognaturel) [reply src/js/relevant.js:211]: `insideRepeat` should be used here (it is equivalent to `repeatPath != null && p.path.startsWith(`${repeatPath}`);`).

**Comment by lognaturel** [comment]
Closing in favor of minor changes at #909. Opening another PR instead of pushing to this one to preserve history.

Since @eyelidlessness is out, I'll let #909 sit for ~a day to see if there are any comments and then will merge and release so we can fix the regression.

**Pull request closed without merging**